cub/device/dispatch/dispatch_batch_memcpy.cuh

```diff
@@ -0,0 +1,631 @@
+                                                         BlevBufferDstsOutItT,
+                                                         BlevBufferSizesOutItT,
+                                                         BlevBufferTileOffsetsOutItT,
+                                                         BlockOffsetT,
+                                                         BLevBufferOffsetTileState,
+                                                         BLevBlockOffsetTileState>;
+
+    auto multi_block_memcpy_kernel = MultiBlockBatchMemcpyKernel<MaxPolicyT,
+                                                                 BufferOffsetT,
+                                                                 BlevBufferSrcsOutItT,
+                                                                 BlevBufferDstsOutItT,
+                                                                 BlevBufferSizesOutItT,
+                                                                 BlevBufferTileOffsetsOutItT,
+                                                                 BLevBufferOffsetTileState,
+                                                                 BlockOffsetT>;
+
+    constexpr uint32_t BLEV_BLOCK_THREADS = ActivePolicyT::AgentLargeBufferPolicyT::BLOCK_THREADS;
+
+    // Get device ordinal
+    int device_ordinal;
+    if (CubDebug(error = cudaGetDevice(&device_ordinal)))
+      return error;
+
+    // Get SM count
+    int sm_count;
+    if (CubDebug(error = cudaDeviceGetAttribute(&sm_count, cudaDevAttrMultiProcessorCount, device_ordinal)))
+      return error;
+
+    // Get SM occupancy for the batch memcpy block-level buffers kernel
+    int batch_memcpy_blev_occupancy;
+    if (CubDebug(error = MaxSmOccupancy(batch_memcpy_blev_occupancy, multi_block_memcpy_kernel, BLEV_BLOCK_THREADS)))
+      return error;
+
+    uint32_t batch_memcpy_blev_grid_size = sm_count * batch_memcpy_blev_occupancy;
+
+    // Construct the tile status for the buffer prefix sum
+    BLevBufferOffsetTileState buffer_scan_tile_state;
+    if (CubDebug(error = buffer_scan_tile_state.Init(num_tiles,
+                                                     allocations[MEM_BLEV_BUFFER_SCAN_STATE],
+                                                     allocation_sizes[MEM_BLEV_BUFFER_SCAN_STATE])))
+    {
+      return error;
+    }
+
+    // Construct the tile status for thread blocks-to-buffer-assignment prefix sum
+    BLevBlockOffsetTileState block_scan_tile_state;
+    if (CubDebug(error = block_scan_tile_state.Init(num_tiles,
+                                                    allocations[MEM_BLEV_BLOCK_SCAN_STATE],
+                                                    allocation_sizes[MEM_BLEV_BLOCK_SCAN_STATE])))
+    {
+      return error;
+    }
+
+    // Invoke init_kernel to initialize buffer prefix sum-tile descriptors
+    THRUST_NS_QUALIFIER::cuda_cub::launcher::triple_chevron(init_grid_size, INIT_KERNEL_THREADS, 0, stream)
```

Comment:
Is there a reason to have this kernel as a separate one? I guess it could be fused with the successive `init_block_scan_states_kernel` into a single kernel.